src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
+                            }
+                        });
+        JsonArray listResp = listRespFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        Assertions.assertTrue(listResp.isEmpty());
+        CountDownLatch latch = new CountDownLatch(2);
+
+        // Create two new recordings
+        CompletableFuture<Void> createRecordingFuture1 = new CompletableFuture<>();
+        MultiMap form1 = MultiMap.caseInsensitiveMultiMap();
+        form1.add("recordingName", "Recording1");
+        form1.add("duration", "5");
+        form1.add("events", "template=ALL");
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form1,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture1)) {
+                                createRecordingFuture1.complete(null);
+                                latch.countDown();
+                            }
+                        });
+        createRecordingFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        CompletableFuture<Void> createRecordingFuture2 = new CompletableFuture<>();
+        MultiMap form2 = MultiMap.caseInsensitiveMultiMap();
+        form2.add("recordingName", "Recording2");
+        form2.add("duration", "5");
+        form2.add("events", "template=ALL");
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form2,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture2)) {
+                                createRecordingFuture2.complete(null);
+                                latch.countDown();
+                            }
+                        });
+        latch.await(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        // Verify recordings created
+        CompletableFuture<JsonArray> listRespFuture2 = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture2)) {
+                                listRespFuture2.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        listResp = listRespFuture2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(
+                "list should have size 2 after recording creation",
+                listResp.size(),
+                Matchers.equalTo(2));
+
+        // GraphQL Query to filter Active recordings by names
+        CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
+        String query =
+                "query { targetNodes {"
```

Comment:
Why not do `targetNodes(filter: { name: "some-node-name" })` here? This way the query looks only for active recordings on the specified target node. I think you could then remove the previous `/api/v1/targets/%s/recordings` query that checks the same data that this query does, and you would not really need the `for (TargetNode targetNode : graphqlResp.data.targetNodes)` loop lower down either.

Reply:
I think that will be filtering node name instead of recording name? Also when I do that, it returns nothing with the logic of creating active recording that I used.

Reply:
That's right, it will filter the nodes, which you can do first and then on the result filter the recording name. The point would be that this way you only get the query response from that particular node - the one on which you started your test recordings - rather than a query response of all of the recordings on every node.